### PR TITLE
clang++: add page

### PR DESCRIPTION
**Pull request by pixelcmtd**
- [x] The page (if new), does not already exist in the repository.
- [x] The page is in the correct platform directory (`common/`, `linux/`, etc.)
- [x] The page has 8 or fewer examples.
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).
- [x] The page follows the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The page description includes a link to documentation or a homepage (if applicable).

### CODE REVIEW

File: pages/common/clang++.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# clang++
+
+> Compiles C++ source files.
+> Part of LLVM.
+> More information: <https://clang.llvm.org>.
+
+- Compile a source code file into an executable binary:
+
+`clang++ {{source.cpp}} -o {{output_executable}}`
```
Comment (CleanMachine1) [inline pages/common/clang++.md:9]: Change throughout
```suggestion
`clang++ {{path/to/source.cpp}} -o {{path/to/output_executable}}`
```

**Review by CleanMachine1 — APPROVED** [review]
After my comment has been addressed then I approve!

File: pages/common/clang++.md (hunk 1 above)
Comment (pixelcmtd) [reply pages/common/clang++.md:9]: Is there any reason to use this syntax here, but not in `clang`, `gcc`, or `g++`?

Comment (CleanMachine1) [reply pages/common/clang++.md:9]: Probably a rule which wasn't used at the time.

Comment (pixelcmtd) [reply pages/common/clang++.md]: Then we should fix this for all the other pages, too.

**Review by navarroaxel — APPROVED** [review]

File: pages/common/clang++.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+- Compile a source code file into an executable binary:
+
+`clang++ {{path/to/source.cpp}} -o {{path/to/output_executable}}`
+
+- Display (almost) all errors and warnings:
+
+`clang++ {{path/to/source.cpp}} -Wall -o {{path/to/output_executable}}`
+
+- Choose a language standard to compile for:
```
Comment (sbrl) [inline pages/common/clang++.md:15]: Perhaps this will read better?
```suggestion
- Choose a language standard to compile with:
```

**Review by sbrl — APPROVED** [review]
Looks good to me! Just got 1 minor change below for you.

**Review by blueskyson — APPROVED** [review]

**Pull request merged**